### PR TITLE
Add labels

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: projects/dotcms-field-elements/src/components/dot-key-value/key-value-table/key-value-table.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-import { DotKeyValueField, DotLabel } from '../../../models';
-import { getTagLabel } from '../../../utils';
+import { DotKeyValueField } from '../../../models';
 
 @Component({
     tag: 'key-value-table'
 })
 export class KeyValueTableComponent {
     @Prop() items: DotKeyValueField[] = [];
     @Prop() disabled = false;
```
Comment (alfredo-dotcms) [inline projects/dotcms-field-elements/src/components/dot-key-value/key-value-table/key-value-table.tsx]: doc?

File: projects/dotcms-field-elements/src/components/dot-label/dot-label.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                <span class="dot-label__text">
+                    {this.label}
+                    <span class="dot-label__required-mark">*</span>
+                </span>
+                <slot />
+            </label>
+        ) : (
+            <label class="dot-label" id={getLabelId(this.name)}>
+                <span class="dot-label__text">{this.label}</span>
+                <slot />
```
Comment (alfredo-dotcms) [inline projects/dotcms-field-elements/src/components/dot-label/dot-label.tsx]: what is this for?

File: projects/dotcms-field-elements/src/components/dot-label/dot-label.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
+import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
```
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-label/dot-label.e2e.ts]: something missing here ? textarea tests in the `dot-label.e2e.ts` ?

File: projects/dotcms-field-elements/src/components/dot-label/dot-label.tsx (hunk 2 above)
Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-label/dot-label.tsx]: All the components used that way now, like here: https://github.com/dotCMS/core-web/pull/1027/files#diff-8f6399efcc455b6c1b3734aa1384eca1R101

File: projects/dotcms-field-elements/src/components/dot-label/dot-label.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    /** (optional) Text to be rendered */
+    @Prop() label = '';
+
+    /** (optional) Determine if it is mandatory */
+    @Prop() required = false;
+
+    render() {
+        return this.required ? (
+            <label class="dot-label" id={getLabelId(this.name)}>
```
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-label/dot-label.tsx]: try to re-use, `dot-label` and `dot-label__text`.

Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-label/dot-label.tsx]: try to re-use, `dot-label` and `dot-label__text` in the condition.

File: projects/dotcms-field-elements/src/components/dot-label/dot-label.e2e.ts (hunk 3 above)
Comment (hmoreras) [inline projects/dotcms-field-elements/src/components/dot-label/dot-label.e2e.ts]: something missing here ? textarea tests in the dot-label.e2e.ts ?

Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-label/dot-label.e2e.ts]: Copy paste.

File: projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx [hunk 5]
```diff
@@ -88,33 +81,29 @@ export class DotCheckboxComponent {
...
-                                checked={this.value.indexOf(trimmedValue) >= 0 || null}
-                                onInput={(event: Event) => this.setValue(event)}
-                                value={trimmedValue}
-                            />
-                            {getTagLabel(labelTagParams)}
-                        </Fragment>
-                    );
-                })}
-                {getTagHint(this.hint)}
+                <dot-label label={this.label} required={this.required} name={this.name}>
```
Comment (freddyucv) [inline projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx]: why not just change the 'getTagLabel'?

File: projects/dotcms-field-elements/src/components/dot-key-value/key-value-table/key-value-table.tsx (hunk 1 above)
Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-key-value/key-value-table/key-value-table.tsx]: When we start doing test and stuff, let's make sure we do docs, params and tests in each component we assigned to.

File: projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx (hunk 5 above)
Comment (fmontes) [reply projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx]: Because we can't use `<slot />` in a function.

**Pull request merged**